go/types/opcache.go

```diff
@@ -8,15 +8,33 @@ import (
 	"encoding/binary"
 	"io/ioutil"
 	"os"
+	"sync"
+	"sync/atomic"
 
 	"github.com/attic-labs/noms/go/d"
 	"github.com/attic-labs/noms/go/hash"
 	"github.com/syndtr/goleveldb/leveldb"
 	"github.com/syndtr/goleveldb/leveldb/iterator"
 	"github.com/syndtr/goleveldb/leveldb/opt"
+	"github.com/syndtr/goleveldb/leveldb/util"
 )
 
-func newOpCache(vrw ValueReadWriter) *opCache {
+var (
+	collectionId = uint32(0)
+	ldb          *opCacheDb
```

Comment:
I'm generally not a fan of the 'Singleton' pattern. No clear ownership tends to lead to failures to clean up and/or shutdown ordering weirdness. What was the issue with each mutator having its own opcache? Too many files open or something?